include/TradingEngine/limit_order.h

```diff
@@ -0,0 +1,54 @@
+/*
+Contains declaration of composition class responsible for managing objects of Order
+class through a doubly LinkedList. The LinkedList ensures that Orders are added and removed
+in accordance with the FIFO (first-in, first-out) principle. New Orders are added to the tail
+of the LinkedList, while Order matching is performed from the head, maintaining the
+desired order.*/
+
+#ifndef TRADING_ENGINE_SOURCE_HEADERS_LIMITORDER_H_
+#define TRADING_ENGINE_SOURCE_HEADERS_LIMITORDER_H_
+
+#pragma once
+#include <string>
+#include <memory>
+#include <vector>
+#include <unordered_set>
+
+#include "transactions.h"
+#include "order_type.h"
+#include "order_status.h"
+#include "order.h"
+
+class LimitOrder {
+private:
+	std::shared_ptr<Order> limit_order_head;
+	std::shared_ptr<Order> limit_order_tail;
+	int limit_order_length;
+
+public:
+
+	LimitOrder() : limit_order_head(nullptr), limit_order_tail(nullptr), limit_order_length(0) {};
+
+	const std::shared_ptr<Order>& getHead();
+	void setHead(const std::shared_ptr<Order>& incoming_order);
+
+	const std::shared_ptr<Order>& getTail();
+	void setTail(const std::shared_ptr<Order>& incoming_order);
+
+	int getListLength() const;
+
+	int updateListLength(const int& update_increment);
```

Comment:
Setters and getters are supposed to be exposed to external classes and methods to use outside your class.

Inside the context of the class itself, you can access the private member functions directly.

It's dangerous to expose a method like this because if an object of this class was used to call that method, you can end up with misleading length.

Reply:
I changed the code to access the private member functions directly. As far as the setter and getters are concerned, at this point I think it would take too much time to refactor the code. So I will keep the setter and getter point for future reference.